guides/gatherers.md

```diff
@@ -269,13 +269,21 @@ Example output (in Rhai):
 
 **Argument required**: yes.
 
-This gatherer returns the version as a string of the specified rpm package.
+This gatherer supports two usecases:
+ - it can be used to return the version as a string of the specified package.
+ - it can be used to compare a given version string against the installed version string of a given package.
+
+While for the first usecase, a simple string containing the version is returned, for the second usecase, the return value is as follows:
+ - A value of `0` if the provided version string matches the installed package version for the requested package.
+ - A value of `-1` if the provided version string is older that what's currently installed.
+ - A value of `1` if the provided version string is newer than what's currently installed.
```

Comment:
we can mention the rpmvercmp algorithm somewhere here (with a link to the algorithm i.e. https://fedoraproject.org/wiki/Archive:Tools/RPM/VersionComparison#The_rpmvercmp_algorithm) as the user might be familiar with it